subprojects/architecture-test/src/test/java/org/gradle/architecture/test/InternalNullabilityTest.java

```diff
@@ -0,0 +1,664 @@
+        "org.gradle.platform.base.internal.dependents",
+        "org.gradle.platform.base.internal.registry",
+        "org.gradle.platform.base.internal.toolchain",
+        "org.gradle.platform.base.plugins",
+        "org.gradle.platform.internal",
+        "org.gradle.plugin.devel.plugins",
+        "org.gradle.plugin.internal",
+        "org.gradle.plugin.management.internal",
+        "org.gradle.plugin.management.internal.autoapply",
+        "org.gradle.plugin.use.resolve.internal",
+        "org.gradle.plugin.use.resolve.internal.local",
+        "org.gradle.plugin.use.resolve.service.internal",
+        "org.gradle.plugin.use.tracker.internal",
+        "org.gradle.plugins.ear",
+        "org.gradle.plugins.ear.descriptor.internal",
+        "org.gradle.plugins.ear.internal",
+        "org.gradle.plugins.ide.api",
+        "org.gradle.plugins.ide.eclipse",
+        "org.gradle.plugins.ide.eclipse.internal",
+        "org.gradle.plugins.ide.eclipse.model",
+        "org.gradle.plugins.ide.eclipse.model.internal",
+        "org.gradle.plugins.ide.idea",
+        "org.gradle.plugins.ide.idea.internal",
+        "org.gradle.plugins.ide.idea.model",
+        "org.gradle.plugins.ide.idea.model.internal",
+        "org.gradle.plugins.ide.internal",
+        "org.gradle.plugins.ide.internal.configurer",
+        "org.gradle.plugins.ide.internal.generator",
+        "org.gradle.plugins.ide.internal.generator.generator",
+        "org.gradle.plugins.ide.internal.resolver",
+        "org.gradle.plugins.ide.internal.resolver.model",
+        "org.gradle.plugins.ide.internal.tooling",
+        "org.gradle.plugins.ide.internal.tooling.eclipse",
+        "org.gradle.plugins.ide.internal.tooling.idea",
+        "org.gradle.plugins.ide.internal.tooling.java",
+        "org.gradle.plugins.ide.internal.tooling.model",
+        "org.gradle.plugins.signing",
+        "org.gradle.plugins.signing.internal",
+        "org.gradle.plugins.signing.signatory.internal",
+        "org.gradle.plugins.signing.signatory.internal.gnupg",
+        "org.gradle.plugins.signing.signatory.internal.pgp",
+        "org.gradle.problems",
+        "org.gradle.problems.buildtree",
+        "org.gradle.problems.internal",
+        "org.gradle.process.internal",
+        "org.gradle.process.internal.health.memory",
+        "org.gradle.process.internal.shutdown",
+        "org.gradle.process.internal.streams",
+        "org.gradle.process.internal.util",
+        "org.gradle.process.internal.worker",
+        "org.gradle.process.internal.worker.child",
+        "org.gradle.process.internal.worker.messaging",
+        "org.gradle.process.internal.worker.request",
+        "org.gradle.reporting",
+        "org.gradle.scripts",
+        "org.gradle.security.internal",
+        "org.gradle.security.internal.gnupg",
+        "org.gradle.security.internal.pgp",
+        "org.gradle.swiftpm.internal",
+        "org.gradle.swiftpm.plugins",
+        "org.gradle.test.fixtures",
```

Comment:
:question: Shouldn't test fixtures be excluded from what we call "internal api"?

Reply:
Yes, I think we should exclude them, given that most of them are actually written in Groovy, the strictness there does not make much sense.